File: src/main/java/pl/dmichalski/algorithms/_1_sum_from_1_to_n/SumCounterService1.java
package pl.dmichalski.algorithms._1_sum_from_1_to_n;

class SumCounterService1 {

    /**
     * O(n) complexity
     */
    public int countSumFrom1ToN(int n) {
        int sum = 0;

        for (int i = 1; i <= n; i++) {
            sum += i;
        }

        return sum;
    }

}

File: src/main/kotlin/pl/dmichalski/algorithms/_1_sum_from_1_to_n/SumCounterService1.kt
package pl.dmichalski.algorithms._1_sum_from_1_to_n

internal class SumCounterService1 {

    /**
     * O(n) time complexity
     */
    fun countSumFrom1ToN(n: Int): Int {
        var sum = 0
        for (i in 1..n) {
            sum += i
        }
        return sum
    }
}
